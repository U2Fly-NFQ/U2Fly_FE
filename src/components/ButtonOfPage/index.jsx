import React from 'react'
import './index.scss'
export default function ButtonOfPage({ title, width, height }) {
  return (
    <div className="ButtonOfPage">
      <button
        style={{ width: width && 'fitContent', height: height && 'fitContent' }}
      >
        {' '}
        {title}
      </button>
    </div>
  )
}
