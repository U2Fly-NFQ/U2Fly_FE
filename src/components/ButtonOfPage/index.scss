@use '../../assets/scss/abstracts/' as a;
.form-submit {
  display: flex;
  // justify-content: center;
  // padding-top: 20px;

  .btn {
    display: inline-block;
    font-weight: 400;
    line-height: 1.5;
    text-align: center;
    text-decoration: none;
    vertical-align: middle;
    cursor: pointer;
    user-select: none;
    font-size: 16px;
    border-radius: 5px;
    box-shadow: none;
    overflow: hidden;
    white-space: nowrap;
    position: relative;
    z-index: 0;

    &.btn-primary {
      color: a.$white-color;
      background-color: a.$main-color;
      transition: a.$transition;
      overflow: hidden;
      white-space: nowrap;
      position: relative;
      z-index: 0;
      border: none;
    }

    &.btn-md {
      padding: 12px 35px;
      font-size: 18px;
    }

    &::before {
      content: '';
      position: absolute;
      top: 0;
      bottom: 0;
      left: 50%;
      width: 580px;
      height: 550px;
      margin: auto;
      background: a.$black-color;
      border-radius: 50%;
      z-index: -1;
      transform-origin: top center;
      transform: translateX(-50%) translateY(-5%) scale(0.4);
      transition: a.$transition;
    }

    &:hover::before {
      transition: transform 1s;
      transform: translateX(-50%) translateY(0) scale(1);
      transform-origin: bottom center;
      border: none;
    }
  }
}
