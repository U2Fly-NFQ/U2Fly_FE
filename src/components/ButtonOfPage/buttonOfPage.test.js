import React from 'react'
import renderer from 'react-test-renderer'

import ButtonOfPage from './'

describe('Button of page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ButtonOfPage />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
