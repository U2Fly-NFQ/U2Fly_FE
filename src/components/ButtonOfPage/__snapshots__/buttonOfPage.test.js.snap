// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button of page renders correctly 1`] = `
<div
  className="form-submit"
>
  <button
    className="btn btn-primary btn-md"
    data-testid="btnCoupon"
    id=""
    type="submit"
  />
</div>
`;
